### ISSUE TITLE
bindgen generates code with reserved rust keywords

### ISSUE THREAD

**Issue by Datamats** [rust-lang-nursery/rust-bindgen#1399]
### Input C/C++ Header

```C++
int test (int async) {

}
```

### Bindgen Invocation

```Rust
bindgen::Builder::default()
    .header("input.h")
    .generate()
    .unwrap()
```

### Actual Results

```
Bindgen itself is fine
```

and/or

```rust
extern "C" { pub fn test ( async : :: std :: os :: raw :: c_int ) -> :: std :: os :: raw :: c_int ; }
```

and/or

```
error: expected pattern, found reserved keyword `async`
 --> //bindings.rs:3:90837
```

### Expected Results

the new rust keywords being mangled

**Comment by emilio** [comment]
`async` should be added to the `rust_mangle` function in `src/ir/context.rs`.

**Comment by emilio** [comment]
We could, yeah...

**Comment by auscompgeek** [comment]
Huh, there's a lot of hard tabs in `rust_mangle`. Is there any particular reason for this?

**Comment by emilio** [comment]
Not really, I think it was introduced accidentally, please do fix it up while you're there if you wish, otherwise I'll do it. Thanks!

**Comment by WizardOfArc** [comment]
I'd like to contribute to this, but as a new to C++ developer (Scala is what I do for work) I'd need a mentor.  Would that be a problem?

**Comment by emilio** [comment]
Nope! I'm happy to mentor as needed, just drop any questions here or in #servo on irc.mozilla.org (I'm `emilio` there as well :))

**Comment by WizardOfArc** [comment]
Thanks @emilio !  I'll ping you on #servo as I think of questions.

**Comment by emilio** [comment]
`async` was fixed in #1405. If anything else is needed other PRs can be fixed the same way. Thanks!

**Comment by nobuyuki** [comment]
Strangely, `async` is included but not `await`.

https://github.com/rust-lang/rust-bindgen/blob/105b9422cd5f8cd7aca9def458c157be219a03ff/src/ir/context.rs#L824
